### DISCUSSION TITLE
Message: Uncaught TypeError: Cannot read property 'DocumentTypeGrid' of undefined

### DISCUSSION THREAD

**Discussion by technet365** [serenity-is/Serenity#5526] [Help]
![Screenshot 2021-02-24 002106](https://user-images.githubusercontent.com/32975430/108915904-36861300-7636-11eb-9a24-c70e002a6bd7.png)

F12 - Debug: 
jquery-3.5.1.js?v=I8fF0tExdQjoB6bH93fW7Q:4046 jQuery.Deferred exception: Cannot read property 'DocumentTypeGrid' of undefined TypeError: Cannot read property 'DocumentTypeGrid' of undefined
    at HTMLDocument.<anonymous> (http://localhost:59952/Documents/DocumentType:214:32)
    at mightThrow (http://localhost:59952/Serenity.Assets/Scripts/jquery-3.5.1.js?v=I8fF0tExdQjoB6bH93fW7Q:3762:29)
    at process (http://localhost:59952/Serenity.Assets/Scripts/jquery-3.5.1.js?v=I8fF0tExdQjoB6bH93fW7Q:3830:12) undefined
jQuery.Deferred.exceptionHook @ jquery-3.5.1.js?v=I8fF0tExdQjoB6bH93fW7Q:4046
jquery-3.5.1.js?v=I8fF0tExdQjoB6bH93fW7Q:4055 Uncaught TypeError: Cannot read property 'DocumentTypeGrid' of undefined
    at HTMLDocument.<anonymous> (DocumentType:214)
    at mightThrow (jquery-3.5.1.js?v=I8fF0tExdQjoB6bH93fW7Q:3762)
    at process (jquery-3.5.1.js?v=I8fF0tExdQjoB6bH93fW7Q:3830)

Sergen: 
Enter a Table: ('!' to abort)
dbo.DocumentState
Enter a Module name for table: ('!' to abort)
Documents
Enter a class Identifier for table: ('!' to abort)
DocumentState
Enter a Permission Key for table: ('!' to abort)
Documents:General
Choose What to Generate (R:Row, S:Repo+Svc, U=UI, C=Custom)
RSUC

Grid file:
namespace SmartOCR.Documents {

    @Serenity.Decorators.registerClass()
    export class DocumentStateGrid extends Serenity.EntityGrid<DocumentStateRow, any> {
        protected getColumnsKey() { return 'Documents.DocumentState'; }
        protected getDialogType() { return DocumentStateDialog; }
        protected getIdProperty() { return DocumentStateRow.idProperty; }
        protected getInsertPermission() { return DocumentStateRow.insertPermission; }
        protected getLocalTextPrefix() { return DocumentStateRow.localTextPrefix; }
        protected getService() { return DocumentStateService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
        }
    }
}

In web.js i don't have anything about DocumentStateGrig.
P.S. DocumentType is in same error.

Thanks for help.

**Comment by VictorTomaili** [comment]
Hello @technet365 it's gives `DocumentTypeGrid` not found because it's was deleted or excluded etc. but `cshtml` still tries load. Your code what generated by sergen is not related `DocumentTypeGrid`. Please check your modules again. And it's normal about you didn't find the `DocumentStateGrig` on your web.js because you get `DocumentTypeGrid` undefined error and you generated `DocumentStateGrid` from sergen. There is nothing named `DocumentStateGrig`

**Comment by technet365** [comment]
sorry for misunderstanding. I have 2 classes, DocumentState and DocumentType. On both classes is same problem.
![Screenshot 2021-02-24 111401](https://user-images.githubusercontent.com/32975430/108977650-6f9fa100-7691-11eb-90b8-ded13a7a8216.png)

**Reply by VictorTomaili** [reply]
I think there is some configuration issue to cause ts files doesn't compile. try to create new project and add this modules. Then run and check for it's works. if is it, check differences between your project and new one.

**Reply by technet365** [reply]
maybe from class declaration?
public class DefaultDB_20210223_000001_DocumentsAtributes : AutoReversingMigration
    {
    public override void Up()
        {
            this.CreateTableWithId32("DocumentType", "DocumentTypeId", s => s
                .WithColumn("DocumentTypeName").AsString(100).NotNullable()
                .WithColumn("DocumentTypeCode").AsString(50).NotNullable());

            Insert.IntoTable("DocumentType").WithIdentityInsert()
                .Row(new
            {

                DocumentTypeId = -110,
                DocumentTypeName = "Invoice",
                DocumentTypeCode = "INV",
            });

            Insert.IntoTable("DocumentType").WithIdentityInsert()
                .Row(new
            {
                DocumentTypeId = -100,
                DocumentTypeName = "Delivery Note",
                DocumentTypeCode = "AVF",
            });
            
            this.CreateTableWithId32("DocumentState", "DocumentStateId", s => s
                .WithColumn("DocumentStateName").AsString(100).NotNullable());

            Insert.IntoTable("DocumentState").WithIdentityInsert()
                .Row(new
            {
                DocumentStateId = 10,
                DocumentStateName = "Draft"
            });
.....

**Reply by technet365** [reply]
Is working in new project. Thank you for help. You can close subject.

**Reply by VictorTomaili** [reply]
Please use discussions for next time. I converted issue to this discussion already.
